db/migrate/20240603204215_add_application_to_reports.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddApplicationToReports < ActiveRecord::Migration[7.1]
+  def change
+    add_column :reports, :application_id, :bigint, null: true
+    add_foreign_key :reports, :oauth_applications, column: :application_id, on_delete: :nullify, validate: false
```

Comment:
Nullify feels like the right option here, since we can't really "preserve" an `oauth_application`, as far as I know?